### DISCUSSION TITLE
Can't write "let" at the start of a line

### DISCUSSION THREAD

**Discussion by StachuDotNet** [darklang/dark#4422] [Suggestions]
<img width="470" alt="Screen Shot 2022-08-24 at 10 29 45 PM" src="https://user-images.githubusercontent.com/906686/186562460-4e63de9e-b558-4588-88b8-5b5e9e218756.png">

Users should be able to quickly wrap the line here with a `let` binding, without interacting with the command dialog.
Is it sufficient to respond on 'L' being pressed, or do we need a more nuanced solution here?

(We may already have an issue around this? didn't check.)

**Comment by pbiggar** [comment]
To solve this, we need to start by describing what the interaction would be. (eg, if we press "l" here, what does the UI show?)

**Reply by StachuDotNet** [reply]
Right - and what else do we respect at this position, besides 'l'?

A user may similarly reach to update a line like this from `A` to `userFn A`, etc. - we should probably solve for the more general case, rather than focus solely on 'let' wrapping here?

**Reply by pbiggar** [reply]
This would be another place where looking at existing structured editors would be helpful.
